cesiumTestsExtension/public/CesiumTests.cpp

```diff
@@ -0,0 +1,93 @@
+#define CARB_EXPORTS
+#define DOCTEST_CONFIG_IMPLEMENT
+#define DOCTEST_CONFIG_IMPLEMENTATION_IN_DLL
+#define DOCTEST_CONFIG_SUPER_FAST_ASSERTS
+
+#include "CesiumTests.h"
+
+#include "cesium/omniverse/Context.h"
+
+#include <carb/PluginUtils.h>
+#include <doctest/doctest.h>
+#include <omni/fabric/IFabric.h>
+
+#include <iostream>
+
+static void exampleTest() {
+    // This file is set up such that we can use doctest assertions (e.x. CHECK)
+    // outside of a test suite.
+
+    // NOTE: failed tests will throw a SIGTRAP in order to call the handler
+    // defined below, which will trigger a pause in any running debugger.
+    // Execution can continue normally from there. To see this in action,
+    // modify the CHECK below to fail then run the extension in a debugger
+
+    CHECK(1 + 1 == 2);
+}
+
+static void handler(const doctest::AssertData& ad) {
+    // Handles failed doctest assertions
+
+    using namespace doctest;
+
+    std::cout << Color::LightGrey << skipPathFromFilename(ad.m_file) << "(" << ad.m_line << "): ";
+    std::cout << Color::Red << failureString(ad.m_at) << ": ";
+
+    // handling only normal (comparison and unary) asserts - exceptions-related asserts have been skipped
+    if (ad.m_at & assertType::is_normal) {
+        std::cout << Color::Cyan << assertString(ad.m_at) << "( " << ad.m_expr << " ) ";
+        std::cout << Color::None << (ad.m_threw ? "THREW exception: " : "is NOT correct!\n");
+        if (ad.m_threw)
+            std::cout << ad.m_exception;
+        else
+            std::cout << "  values: " << assertString(ad.m_at) << "( " << ad.m_decomp << " )";
+    } else {
+        std::cout << Color::None << "an assert dealing with exceptions has failed!";
+    }
+
+    std::cout << std::endl;
+}
```

Comment:
>    // This file is set up such that we can use doctest assertions (e.x. CHECK)
>    // outside of a test suite.

Is there a reason that we're running doctests assertions outside of a test suite? It would be nice to avoid all this assertion handling.

Reply:
I tried defining a test suite/case previously, but couldn't make it work. Doctest does provide the ability to define test cases within classes (via the TEST_CASE_CLASS macro), but it seems to require instantiating the object, which for the interface class is tricky if it's doable at all. I couldn't even get a simpler test case to compile in these classes though. If I placed a test case inside a method, the compiler bans a function definition that the macro does. If I placed it outside a method and in the header as the example doc shows, I get a compilation error that suggests an internal doctest issue (`error: ‘DOCTEST_ANON_VAR_15’ was not declared in this scope`)

Thankfully the one handler can cover all the tests

Reply:
Sort of a follow up question: is it possible to see a test report similar to what we get with the `tests` executable?

```
[doctest] doctest version is "2.4.9"
[doctest] run with "--help" for options
===============================================================================
[doctest] test cases:   10 |   10 passed | 0 failed | 0 skipped
[doctest] assertions: 1277 | 1277 passed | 0 failed |
[doctest] Status: SUCCESS!
```

Currently, we just see which tests have failed. And without `TEST_SUITE` there's no grouping of tests.

It doesn't necessarily need to hold up the PR, but eventually I would hope for something a bit more ergonomic.

Reply:
agreed